dkc/core/models/folder.py

```diff
@@ -116,6 +122,16 @@ def clean(self) -> None:
             )
         super().clean()
 
+    @classmethod
+    def filter_by_permission(
+        cls, user: User, permission: Permission, queryset: models.QuerySet['Folder']
+    ) -> models.QuerySet['Folder']:
+        tree_query = Tree.filter_by_permission(user, permission, Tree.objects).values('pk')
+        return queryset.filter(tree__in=models.Subquery(tree_query))
```

Comment:
Another non-actionable comment for information. This builds queries of the form:

```python
>>> q = Folder.filter_by_permission(user, Permission.read, Folder.objects)
>>> print(q.query)
SELECT "core_folder"."id", "core_folder"."created", "core_folder"."modified", "core_folder"."name", "core_folder"."depth", "core_folder"."size", "core_folder"."description", "core_folder"."user_metadata", "core_folder"."parent_id", "core_folder"."tree_id" FROM "core_folder" WHERE "core_folder"."tree_id" IN ((SELECT "core_tree"."id" FROM "core_tree") UNION (SELECT "core_tree"."id" FROM "core_tree" WHERE "core_tree"."public")) ORDER BY "core_folder"."name" ASC

>>> q = Folder.filter_by_permission(user, Permission.write, Folder.objects)
>>> print(q.query)
SELECT "core_folder"."id", "core_folder"."created", "core_folder"."modified", "core_folder"."name", "core_folder"."depth", "core_folder"."size", "core_folder"."description", "core_folder"."user_metadata", "core_folder"."parent_id", "core_folder"."tree_id" FROM "core_folder" WHERE "core_folder"."tree_id" IN (SELECT U0."id" FROM "core_tree" U0) ORDER BY "core_folder"."name" ASC
```
```